pybamm/expression_tree/unary_operators.py

```diff
@@ -364,22 +398,43 @@ class Integral(SpatialOperator):
     """
 
     def __init__(self, child, integration_variable):
-        if isinstance(integration_variable, pybamm.SpatialVariable):
-            # Check that child and integration_variable domains agree
-            if child.domain != integration_variable.domain:
-                raise pybamm.DomainError(
-                    """child and integration_variable must have the same domain"""
-                )
-        elif not isinstance(integration_variable, pybamm.IndependentVariable):
-            raise ValueError(
-                """integration_variable must be of type pybamm.IndependentVariable,
-                   not {}""".format(
-                    type(integration_variable)
+        if isinstance(integration_variable, list):
```

Comment:
if integration variable is a symbol, you can convert it to a list with just one element (`integration_variable=[integration_variable]`) and then avoid some repeated code